pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/jaypipes/ghw"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+const (
+	// obtained these values from node e2e tests : https://github.com/kubernetes/kubernetes/blob/82baa26905c94398a0d19e1b1ecf54eb8acb6029/test/e2e_node/util.go#L70
+	defaultPodResourcesTimeout = 10 * time.Second
+)
+
+type nodeResources struct {
+	perNUMACapacity map[int]map[v1.ResourceName]int64
+	// mapping: resourceName -> resourceID -> nodeID
+	resourceID2NUMAID map[string]map[string]int
+	topo              *ghw.TopologyInfo
+}
```

Comment:
This looks kludge-ish. I think we should have the concept of zones in the aggregator, too. In the first phase handle e.g. sockets and numa nodes and then possibly extend into finer details as further iterations. Having "two levels" of zones from the beginning would better evaluate the concept (and the consumers of the data, too)

Reply:
Code refactoring to improve the readability and in order to handle "two levels" of zones  would be handled in a separate follow up PR(s). Please refer to the comment regarding this here: https://github.com/kubernetes-sigs/node-feature-discovery/pull/525#issuecomment-888576969.

Reply:
Let me know if that is okay.